misc/python/materialize/checks/all_checks/alter_index.py

```diff
@@ -9,14 +9,15 @@
 from textwrap import dedent
 
 from materialize.checks.actions import Testdrive
-from materialize.checks.checks import Check
+from materialize.checks.checks import Check, externally_idempotent
 from materialize.checks.common import KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD
 
 
 def schema() -> str:
     return dedent(KAFKA_SCHEMA_WITH_SINGLE_STRING_FIELD)
 
 
+@externally_idempotent(False)
```

Comment:
@def- @nrainer-materialize I am not very confident about this -- I had to decorate one dozen Checks with this , that is, all Checks that modify external state in their manipulate(). This means that going forward, if one adds a new Check that is not compatible, it will fail in Nightly in this particular backup+restore scenario and then one needs to manually figure out if it is a backup+restore problem or it is just that the check had to be decorated with `externally_idempotent`. This sounds like maintenance nightmare going forward ...

One option would be to wait for the backup+restore functionality to advance beyond the prototype stage before we add more testing for it, especially if it is going to be disruptive going forward as described above.

Reply:
I like this, even though it's extra effort. It gives us another dimension on which we can test platform-checks. Seems worth the effort.